### PR TITLE
Bitfield cleanup

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/login/logind-session.h [hunk 1]
```diff
@@ -135,16 +135,17 @@ struct Session {
         sd_event_source *fifo_event_source;
         sd_event_source *leader_pidfd_event_source;
 
-        bool idle_hint;
-        dual_timestamp idle_hint_timestamp;
 
         bool locked_hint;
```
Comment (YHNdnzj) [inline src/login/logind-session.h]: (I prefer to place this close to idle_hint, below `started` and friends)

File: src/core/kmod-setup.c [hunk 2]
```diff
@@ -166,13 +165,12 @@ int kmod_setup(void) {
...
         };
-        _cleanup_(kmod_unrefp) struct kmod_ctx *ctx = NULL;
-        unsigned i;
 
         if (have_effective_cap(CAP_SYS_MODULE) <= 0)
                 return 0;
 
-        for (i = 0; i < ELEMENTSOF(kmod_table); i++) {
+        _cleanup_(kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        for (unsigned i = 0; i < ELEMENTSOF(kmod_table); i++) {
```
Comment (YHNdnzj) [inline src/core/kmod-setup.c]: This can be switched to FOREACH_ARRAY ;-)

Comment (poettering) [reply src/core/kmod-setup.c]: also, size_t is the right type when iterating through memory, it's also the type ELEMENTSOF() returns, hence it's weird dropping half of the bits on the ground for no real reason

**Comment by keszybz** [comment]
Updated as requested.

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
Looks unrelated but new @mrc0mmand ?

```
[   28.746956] testsuite-53.sh[544]: ++ systemctl show test-timer.timer -p NextElapseUSecRealtime --value
[   29.620869] testsuite-53.sh[540]: + next_elapsed='Tue 2024-02-20 11:00:00 UTC'
[   29.644884] testsuite-53.sh[549]: ++ date -d 'Tue 2024-02-20 11:00:00 UTC' +%s
[   29.689054] testsuite-53.sh[540]: + next_elapsed=1708426800
[   29.716021] testsuite-53.sh[550]: ++ date +%s
[   29.769279] testsuite-53.sh[540]: + now=1708426032
[   29.783497] testsuite-53.sh[540]: + time_delta=768
[   29.792240] testsuite-53.sh[540]: + (( 0 < time_delta && time_delta < 1200 ))
[   29.797396] testsuite-53.sh[540]: + test '!' -s /failed
```

**Pull request merged**